### PR TITLE
Convert returning ok into modern errors

### PR DESCRIPTION
**Pull request by oerc0122**
Update to use modern error handling methods

### CODE REVIEW

File: herbert_core/applications/multifit/@mfclass/fit.m [hunk 1]
```diff
@@ -111,57 +103,45 @@
...
 end
 
 % Check that all functions are present
 foreground_present = ~all(cellfun(@isempty,obj.fun_));
 background_present = ~all(cellfun(@isempty,obj.bfun_));
+
 if ~foreground_present && ~background_present
-    ok = false; mess = 'No fit functions have been provided';
-    if throw_error, error_message(mess), else, return, end
+    error("Herbert:mfclass:badargs", 'No fit functions have been provided')
```
Comment (hsautessella) [inline herbert_core/applications/multifit/@mfclass/fit.m:125]: [ADR 21](https://github.com/pace-neutrons/Horace/blob/master/documentation/adr/0021-errors-and-warnings.md#recommendations) suggests the first two parts of the error ID should be upper case

File: herbert_core/applications/multifit/@mfclass/private/mask_data_for_fit.m [hunk 2]
```diff
@@ -20,60 +20,52 @@
...
-    else
-        wmask=cell(1,0);
-        msk_out=cell(1,0);
-        mess=[data_id_mess(sz,i),mess_tmp];
-        return
-    end
-    % Mask data - only if there is some to be masked (don't want to change array sizes otherwise)
-    if ~all(msk_out{i}(:))
+    for i = 1:numel(w)
+% Accumulate bad points (y = NaN, zero error bars etc.) to the mask array
```
Comment (hsautessella) [inline herbert_core/applications/multifit/@mfclass/private/mask_data_for_fit.m:37]: Lost indentation on this comment

File: herbert_core/applications/multifit/@mfclass/private/mask_data_for_fit.m [hunk 3]
```diff
@@ -20,60 +20,52 @@
...
         else % a different data object
-            wmask{i}=mask(w{i},msk_out{i});
+            [ytmp,vtmp,msk_null] = sigvar_get(w{i});
+            [msk_out{i}, mess_tmp] = mask_points_for_fit_xye({},ytmp,vtmp,(msk_in{i}&msk_null));
+        end
+        if ~isempty(mess_tmp)
+            warning("Herbert:mfclass:maskissue", accumulate_message(mess,[data_id_mess(sz,i),mess_tmp]));
+        end
+
+% Mask data - only if there is some to be masked (don't want to change array sizes otherwise)
```
Comment (hsautessella) [inline herbert_core/applications/multifit/@mfclass/private/mask_data_for_fit.m:48]: Indentation lost on this comment as well

**Review by hsautessella — COMMENTED** [review]
Changes look fine to me - couple of minor things

**Review by nickbattam-tessella — COMMENTED** [review]
Does this change have impact on @cmarooney-stfc updates to fit on SQW?

**Review by abuts — COMMENTED** [review]
Reasonable intention but would be better to follow suggestion above

**Review by cmarooney-stfc — APPROVED** [review]

**Pull request merged**